packages/store-engine-localstorage/src/LocalStorageEngine.ts

```diff
@@ -0,0 +1,213 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {CRUDEngine, error as StoreEngineError} from '@wireapp/store-engine';
+
+export class LocalStorageEngine implements CRUDEngine {
+  private autoIncrementedPrimaryKey: number = 1;
+  public storeName = '';
+
+  public append<PrimaryKey = string>(
+    tableName: string,
+    primaryKey: PrimaryKey,
+    additions: string,
+  ): Promise<PrimaryKey> {
+    return this.read(tableName, primaryKey).then((record: any) => {
+      if (typeof record === 'string') {
+        record += additions;
+      } else {
+        const message = `Cannot append text to record "${primaryKey}" because it's not a string.`;
+        throw new StoreEngineError.RecordTypeError(message);
+      }
+
+      const key = this.createKey<PrimaryKey>(tableName, primaryKey);
+      window.localStorage.setItem(`${key}`, record);
+
+      return primaryKey;
+    });
+  }
+
+  public create<EntityType, PrimaryKey = string>(
+    tableName: string,
+    primaryKey: PrimaryKey,
+    entity: EntityType,
+  ): Promise<PrimaryKey> {
+    if (entity) {
+      const internalPrimaryKey = this.createKey<PrimaryKey>(tableName, primaryKey);
+      return this.read(tableName, primaryKey)
+        .catch(error => {
+          if (error instanceof StoreEngineError.RecordNotFoundError) {
+            return undefined;
+          }
+          throw error;
+        })
+        .then(record => {
+          if (record) {
+            const message = `Record "${primaryKey}" already exists in "${tableName}". You need to delete the record first if you want to overwrite it.`;
+            throw new StoreEngineError.RecordAlreadyExistsError(message);
+          } else {
+            if (typeof record === 'string') {
+              window.localStorage.setItem(`${internalPrimaryKey}`, String(entity));
```

Comment:
why wrapping it in a `String` object?

Reply:
Because otherwise it's of type `PrimaryKey` and the `dom.d.ts` forces me to pass a `string` to `localStorage.setItem`.